BuildNotifications.Core/Config/ConfigurationService.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System.Linq;
+
+namespace BuildNotifications.Core.Config
+{
+    internal class ConfigurationService : IConfigurationService
+    {
+        public ConfigurationService(IConfigurationSerializer serializer, IConfigurationBuilder configurationBuilder)
+        {
+            Serializer = serializer;
+            CurrentConfig = configurationBuilder.LoadConfiguration();
+        }
+
+        private static bool IsSame(IProjectConfiguration x, IProjectConfiguration y) => x.ProjectName == y.ProjectName;
+
+        private static bool IsSame(ConnectionData x, ConnectionData y) => x.ConnectionType == y.ConnectionType && x.Name == y.Name;
+
+        private void MergeConnections(IConfiguration newConfiguration)
+        {
+            foreach (var newConnection in newConfiguration.Connections)
+            {
+                var existingConnection = CurrentConfig.Connections.FirstOrDefault(x => IsSame(x, newConnection));
+                if (existingConnection == null)
+                {
+                    existingConnection = new ConnectionData
+                    {
+                        Name = newConnection.Name,
+                        ConnectionType = newConnection.ConnectionType
+                    };
+
+                    CurrentConfig.Connections.Add(existingConnection);
+                }
+
+                existingConnection.PluginConfiguration = newConnection.PluginConfiguration;
+                existingConnection.PluginType = newConnection.PluginType;
+            }
+        }
+
+        private void MergeProjects(IConfiguration newConfiguration)
+        {
+            foreach (var newProject in newConfiguration.Projects)
+            {
+                var existingProject = CurrentConfig.Projects.FirstOrDefault(x => IsSame(x, newProject));
+                if (existingProject == null)
+                {
+                    existingProject = new ProjectConfiguration
+                    {
+                        ProjectName = newProject.ProjectName
+                    };
+
+                    CurrentConfig.Projects.Add(existingProject);
+                }
+
+                existingProject.DefaultCompareBranch = newProject.DefaultCompareBranch;
```

Comment:
Create a new issue, to use C#9s record for projects? This way we could spare us all this mapping code.

Reply:
Jup sounds good. Created #267 for this